Convert COR Job Unlock and AF1 quests to Interaction Framework

claywar opened the pull request:
<!-- remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm: -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

claywar committed:
Interaction - Quest: Luck of the Draw


Formatting for Luck of the Draw

claywar committed:
Interaction - Quest: Equipped for All Occasions

KnowOne134 commented on scripts/quests/ahtUrhgan/Equipped_for_All_Occasions.lua:
```diff
@@ -0,0 +1,133 @@
+            ['Ratihb'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 4 and quest:getVar(player, 'HasSeenDialog') then
```
GetVar pointing to no number

Xaver-DaRed commented on scripts/zones/Aht_Urhgan_Whitegate/npcs/Ratihb.lua:
```diff
@@ -15,16 +15,7 @@ entity.onTrade = function(player, npc, trade)
 
 entity.onTrigger = function(player, npc)
-    local luckOfTheDraw = player:getQuestStatus(xi.quest.log_id.AHT_URHGAN, xi.quest.id.ahtUrhgan.LUCK_OF_THE_DRAW)
-    local againstAllOdds = player:getQuestStatus(xi.quest.log_id.AHT_URHGAN, xi.quest.id.ahtUrhgan.AGAINST_ALL_ODDS)
```
~~Isn't this variable (againstAllOdds) still needed and used a little down bellow?~~

Actually, the one used is "AgainstAllOdds", and is a Char var.

Xaver-DaRed commented on scripts/zones/Arrapago_Reef/npcs/qm6.lua:
```diff
@@ -19,23 +19,12 @@ end
     local mLvl   = player:getMainLvl()
-    local lotdCS = player:getCharVar("LuckOfTheDraw")
     local efao   = player:getQuestStatus(xi.quest.log_id.AHT_URHGAN, xi.quest.id.ahtUrhgan.EQUIPPED_FOR_ALL_OCCASIONS)
     local efaoCS = player:getCharVar("EquippedforAllOccasions")
```
This variable (efaoCS) isnt used any more

claywar committed:
QC: Fix Luck of the Draw, Equipped for All Occasions


Fix check value


Align with standard quest variable set HasSeenDialog to Stage


Simplify Quest Complete for Luck of the Draw


Fix typo in comment


Remove unused variable in qm6
